pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -417,7 +417,9 @@ public String alterTableStateOrListTableConfig(
   @ApiOperation(value = "Deletes a table", notes = "Deletes a table")
   public SuccessResponse deleteTable(
       @ApiParam(value = "Name of the table to delete", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr) {
+      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr,
+      @ApiParam(value = "Retention period for the table segments (e.g. 12h, 3d); Using 0d or -1d will instantly "
+          + "delete segments without retention") @QueryParam("retention") String retentionPeriod) {
```

Comment:
```suggestion
      @ApiParam(value = "Retention period for the table segments (e.g. 12h, 3d); "
          + " If not set, the retention period will default to cluster setting; Using 0d or -1d will instantly "
          + "delete segments without retention.") @QueryParam("retention") String retentionPeriod) {
```

Reply:
What about the retention period specified in the table? :) 
Perhaps it's simpler to use a boolean flag for whether they want instant delete or not?

Reply:
Retention period on table will not be honored since the table config will be deleted first.